cpp/src/arrow/compute/api_aggregate.h

```diff
@@ -37,43 +37,17 @@ class ExecContext;
 // ----------------------------------------------------------------------
 // Aggregate functions
 
-/// \addtogroup compute-concrete-options
-/// @{
-
-/// \brief Control Count kernel behavior
-///
-/// By default, all non-null values are counted.
-struct ARROW_EXPORT CountOptions : public FunctionOptions {
-  enum Mode {
-    /// Count all non-null values.
-    COUNT_NON_NULL = 0,
-    /// Count all null values.
-    COUNT_NULL,
-  };
-
-  explicit CountOptions(enum Mode count_mode = COUNT_NON_NULL) : count_mode(count_mode) {}
-
-  static CountOptions Defaults() { return CountOptions(COUNT_NON_NULL); }
-
-  enum Mode count_mode;
-};
-
-/// \brief Control MinMax kernel behavior
+/// \brief Control general scalar aggregate kernel behavior
 ///
 /// By default, null values are ignored
-struct ARROW_EXPORT MinMaxOptions : public FunctionOptions {
-  enum Mode {
-    /// Skip null values
-    SKIP = 0,
-    /// Any nulls will result in null output
-    EMIT_NULL
-  };
+struct ARROW_EXPORT ScalarAggregateOptions : public FunctionOptions {
+  explicit ScalarAggregateOptions(bool skip_nulls = true, uint32_t min_count = 1)
+      : skip_nulls(skip_nulls), min_count(min_count) {}
 
-  explicit MinMaxOptions(enum Mode null_handling = SKIP) : null_handling(null_handling) {}
+  static ScalarAggregateOptions Defaults() { return ScalarAggregateOptions{}; }
 
-  static MinMaxOptions Defaults() { return MinMaxOptions{}; }
-
-  enum Mode null_handling;
+  bool skip_nulls = true;
+  uint32_t min_count = 1;
```

Comment:
Was there any former discussion on what would be the best default here? (eg both pandas and R would set this to min_count=0, I think) 
Not that it matters too much as long as it's an option to change.

Reply:
There was no discussion about this that I noticed and you're right 0 seems the better default here. Will change.

Reply:
On the other hand, for mean/min/max, a `min_count=1` might be a more sensible default, to not get NaN/Inf/-Inf for empty/all-null arrays.

(anyway, I don't have a strong opinion on the default, and it probably greatly depends on your exact use case what you find the best default. So as long as the option is there and clearly documented, that should be fine)

Reply:
Switching back to `min_count=0` as it seems to be preferred behavior in c++ (e.g. hash aggregate tests).
Defaults in python and R can still be set differently.